frontend/app-development/features/dataModelling/hooks/useValidateSchemaName.test.ts

```diff
@@ -0,0 +1,120 @@
+import { textMock } from '@studio/testing/mocks/i18nMock';
+import { renderHook, act } from '@testing-library/react';
+import {
+  dataModel1NameMock,
+  jsonMetadata1Mock,
+} from '../../../../packages/schema-editor/test/mocks/metadataMocks';
+import type { ApplicationMetadata } from 'app-shared/types/ApplicationMetadata';
+import { mockAppMetadata } from '../../../layout/PageHeader/SubHeader/SettingsModalButton/SettingsModal/mocks/applicationMetadataMock';
+
+//Test data
+const org = 'test-org';
+const app = 'test-app';
+const dataModels: DataModelMetadata[] = [jsonMetadata1Mock];
+const appMetaData: ApplicationMetadata = mockAppMetadata;
+
+jest.mock('app-shared/hooks/queries', () => ({
+  useAppMetadataQuery: jest.fn(),
+}));
+
+jest.mock('app-shared/hooks/useStudioEnvironmentParams', () => ({
+  useStudioEnvironmentParams: jest.fn(),
+}));
+
+describe('useValidateSchemaName', () => {
+  beforeEach(() => {
+    (useStudioEnvironmentParams as jest.Mock).mockReturnValue({ org, app });
+    (useAppMetadataQuery as jest.Mock).mockReturnValue({ data: appMetaData });
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should set nameError to empty string when name is valid', () => {
+    const { result } = renderHook(() => useValidateSchemaName(dataModels));
+
+    result.current.validateName('ValidName');
+
+    expect(result.current.nameError).toBe('');
+  });
+
+  it('should set error when name is empty', () => {
+    const { result } = renderHook(() => useValidateSchemaName(dataModels));
+
+    act(() => {
+      result.current.validateName('');
+    });
+
+    expect(result.current.nameError).toBe(textMock('validation_errors.required'));
+  });
+
+  it('should set error when name does not match regex', () => {
+    const { result } = renderHook(() => useValidateSchemaName(dataModels));
+
+    act(() => {
+      result.current.validateName('123InvalidName');
+    });
+
+    expect(result.current.nameError).toBe(textMock('schema_editor.error_invalid_datamodel_name'));
+  });
+
+  it('should set error when name exceeds max length', () => {
+    const longName = 'a'.repeat(SCHEMA_NAME_MAX_LENGTH + 1);
+    const { result } = renderHook(() => useValidateSchemaName(dataModels));
+
+    act(() => {
+      result.current.validateName(longName);
+    });
+
+    expect(result.current.nameError).toBe(
+      textMock('validation_errors.maxLength', { 0: SCHEMA_NAME_MAX_LENGTH }),
+    );
+  });
+
+  it('should set error when name exists in dataModels', () => {
+    const { result } = renderHook(() => useValidateSchemaName(dataModels));
+
+    act(() => {
+      result.current.validateName(dataModel1NameMock);
+    });
+
+    expect(result.current.nameError).toBe(
+      textMock('schema_editor.error_model_name_exists', {
+        newModelName: dataModel1NameMock,
+      }),
+    );
+  });
+
+  it('should set error when name exists in dataTypes', () => {
+    const { result } = renderHook(() => useValidateSchemaName(dataModels));
+
```

Comment:
If we user renderWithProviders, we can write this one like this:
```
const queryClient = createQueryClientMock();
    queryClient.setQueryData([QueryKey.AppMetadata, org, app], appMetaData);
    const { result } = renderHookWithProviders(() => useValidateSchemaName(dataModels), {queryClient});

```
Then it is also easy to see which test that is depended on what data 😄

Reply:
I tried to use this, but it fails on one of the tests. Do you have any tips?
```
it('should set error when name exists in dataTypes', () => {
    const { result } = renderUseValidateSchemaName();

    act(() => {
      result.current.validateName(mockAppMetadata.dataTypes[0].id);
    });

    expect(result.current.nameError).toBe(textMock('schema_editor.error_data_type_name_exists'));
  });
```
```
const renderUseValidateSchemaName = () => {
  const queryClient = createQueryClientMock();
  queryClient.setQueryData([QueryKey.AppMetadata, org, app], mockAppMetadata);
  const { renderHookResult: result } = renderHookWithProviders(
    {},
    queryClient,
  )(() => useValidateSchemaName(dataModels));
  return result;
};
```
```
Error: expect(received).toBe(expected) // Object.is equality

Expected: "[mockedText(schema_editor.error_data_type_name_exists)]"
Received: ""
```

Reply:
Fixed 😊